README.md

```diff
@@ -62,6 +65,9 @@ Seq.of(1, 2, 3, 4).grouped(i -> i % 2);
 // (tuple(1, 1), tuple(2, 2))
 Seq.of(1, 2, 4).innerJoin(Seq.of(1, 2, 3), (a, b) -> a == b);
 
+// (tuple(1, 1), tuple(2, 2))
+Seq.of(1, 2).innerSelfJoin((t, u) -> t == u)

```

Comment:
Hmm, this would be a rather inefficient alternative to:

```java
Seq.of(1, 2).map(i -> tuple(i, i))
```

Perhaps, a more convincing example might be interesting here, similar to your outer join examples, where child / parent relationships are joined. What do you think?